lib/adequate_crypto_address/base.rb

```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module AdequateCryptoAddress
+  class Base
+    attr_reader :address
+
+    def initialize(address_string)
+      @address = address_string
+    end
+
+    def valid?(address)
+      valid_prefix?
```

Comment:
trying to figure out a way to do `address.respond_to?(:valid_length?) then valid_length?` bc not every class inheriting Base is checking for length

Reply:
You can add `valid_length?` method to the base class with default of `true`. Then override for the ones that need it.